packages/api/tests/endpoints/test-granules.js

```diff
@@ -1585,3 +1642,138 @@ test.serial('POST return bad request if a granule is submitted with a bad collec
   t.is(response.error.status, 400);
   t.is(response.error.message, 'cannot POST /granules (400)');
 });
+
+test.serial('update (PUT) returns bad request if granule does not exist', async (t) => {
+  const newGranule = fakeGranuleFactoryV2({
+    collectionId: t.context.collectionId,
+    execution: undefined,
+  });
+
+  const response = await request(app)
+    .put(`/granules/${newGranule.granuleId}`)
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .set('Accept', 'application/json')
+    .send(newGranule)
+    .expect(400);
+
+  t.is(response.body.error, 'Bad Request');
+  t.is(response.body.message, `No granule found to update for ${JSON.stringify(newGranule)}`);
+});
+
+test.serial('update (PUT) returns an updated granule with an undefined execution', async (t) => {
+  const now = Date.now();
+  const newGranule = fakeGranuleFactoryV2({
+    collectionId: t.context.collectionId,
+    createdAt: now,
+    timestamp: now,
+    execution: undefined,
+  });
+
+  const response = await request(app)
+    .post('/granules')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .set('Accept', 'application/json')
+    .send(newGranule)
+    .expect(200);
+
+  t.is(response.statusCode, 200);
+
+  const modifiedGranule = {
+    ...newGranule,
+    status: 'failed',
+    error: { some: 'error' },
+  };
+
+  const modifiedResponse = await request(app)
+    .put(`/granules/${newGranule.granuleId}`)
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .set('Accept', 'application/json')
+    .send(modifiedGranule)
+    .expect(200);
+
+  t.is(modifiedResponse.statusCode, 200);
+
+  const fetchedDynamoRecord = await granuleModel.get({
+    granuleId: newGranule.granuleId,
+  });
+
+  const fetchedPostgresRecord = await granulePgModel.get(
+    t.context.knex,
+    {
+      granule_id: newGranule.granuleId,
+      collection_cumulus_id: t.context.collectionCumulusId,
+    }
+  );
+
+  t.deepEqual(JSON.parse(modifiedResponse.text), {
+    message: `Successfully updated granule with Granule Id: ${newGranule.granuleId}`,
+  });
+
+  t.is(fetchedDynamoRecord.status, 'failed');
+  t.deepEqual(fetchedDynamoRecord.error, { some: 'error' });
+  t.is(new Date(fetchedDynamoRecord.timestamp).valueOf(), now);
+  t.is(new Date(fetchedDynamoRecord.createdAt).valueOf(), now);
+  t.is(fetchedPostgresRecord.status, 'failed');
+  t.deepEqual(fetchedPostgresRecord.error, { some: 'error' });
+  t.is(new Date(fetchedPostgresRecord.timestamp).valueOf(), now);
+  t.is(new Date(fetchedPostgresRecord.created_at).valueOf(), now);
+});
+
+test.serial('update (PUT) returns an updated granule with associated execution', async (t) => {
+  const now = Date.now();
+  const newGranule = fakeGranuleFactoryV2({
+    collectionId: t.context.collectionId,
+    createdAt: now,
+    timestamp: now,
+    execution: t.context.executionUrl,
```

Comment:
Can you move `execution: t.context.executionUrl,` to `modifiedGranule`?  Then we can check if fetchedDynamoRecord and fetchedPostgresRecord have execution info.

Reply:
It's spread into the modified granule.  I can add a test for it in the dynamo record, but it isn't returned in the PostgreSQL record.
Are you saying update the modifiedGranule to have a different execution?

Reply:
Ah, take it out of the first granule and move it to the modified and make sure it's updated in both?

Reply:
yeap, to make it simple. just move it to the modified granule
I thought PostgreSQL record has execution info.

Reply:
3dd344a6e2

Reply:
3dd344a6e2

Reply:
👍